### DISCUSSION TITLE
Match threshold selection tool uses scientific notation, not ordinary numbers

### DISCUSSION THREAD

**Discussion by aalexandersson** [moj-analytical-services/splink#2112] [Q&A] [answered]
The [threshold_selection_tool_from_labels_table](https://moj-analytical-services.github.io/splink/charts/threshold_selection_tool_from_labels_table.html) axes use ordinary numbers according to the link. I can reproduce those results. On my test data, however, the tool axes numbers are in scientific notation, not ordinary numbers. Here is an example chart:
 
![c_threshold_nossn](https://github.com/moj-analytical-services/splink/assets/20467610/4cf30769-6f60-428c-b3ba-ec56f7476b5f)
 
For example, the scale shows -1e+1 and +1e+1 but I instead want -10 and +10. I am using Splink 3.9.14 on Windows 10. 

is this a known issue?

Update: I guess that the issue is this code in [files/chart_defs/threshold_selection_tool.json](https://github.com/moj-analytical-services/splink/blob/5d7ea7c913c3dbf29f8324d298bf87cc2b0769d8/splink/files/chart_defs/threshold_selection_tool.json):

```
        "axisX": {
            "offset": 20,
            "format": "+",
            "values": {
                "expr": "[-25,-20,-15,-10,-5,0,5,10,15,20,25]"
            },
            "grid": false
        },
```
My test data are outside the range in the x-axis.

**Comment by aalexandersson** [comment]
I can create the wanted graph using the Vega Editor. If the [**D3 format**](https://observablehq.com/@d3/d3-format?collection=@d3/d3-format) is "+.0f" instead of merely "+", then the tool uses ordinary numbers also for my graph with more extreme values. How can I get this wanted **format** without having to use Vega Editor? Here is the PNG graph I want, in terms of the format:

![c_threshold_edited](https://github.com/moj-analytical-services/splink/assets/20467610/01b721fc-0a33-4271-ac2e-c754899e56b4)

**Reply by RobinL** [reply]
Thanks for investigating this -  it sounds like we need to update the chart definition that you found at [files/chart_defs/threshold_selection_tool.json](https://github.com/moj-analytical-services/splink/blob/master/splink/files/chart_defs/threshold_selection_tool.json)  to be `+.0f` instead of just `+`.  Would you like to do a PR to make the change?  It's such a small change you should be able to do it using the 'edit this file button: 
<img width="197" alt="image" src="https://github.com/moj-analytical-services/splink/assets/2608005/ddd83f2f-ccb2-4971-bd92-a8fa9302e92f">

**Reply by aalexandersson** [reply, marked as answer]
The button is not available to me but I was able to submit  [PR #2120](https://github.com/moj-analytical-services/splink/pull/2120).
